src/model_repository_manager/model_repository_manager.cc

```diff
@@ -326,64 +323,131 @@ GetDetailedModifiedTime(
   if (!status.IsOk()) {
     LOG_ERROR << "Failed to determine modification time for '" << model_dir_path
               << "': " << status.AsString();
-    return std::make_pair(0, 0);
+    return;
   }
   if (!is_dir) {
     LOG_ERROR << "Failed to determine modification time for '" << model_dir_path
               << "': Model directory path is not a directory";
-    return std::make_pair(0, 0);
+    return;
   }
 
-  std::pair<int64_t, int64_t> mtime(0, 0);  // <config.pbtxt, model files>
-
-  // Populate 'model files' time to the model directory time
-  status = FileModificationTime(model_dir_path, &mtime.second);
+  // Populate time for 'model_dir_path'.
+  int64_t model_dir_time = 0;
+  status = FileModificationTime(model_dir_path, &model_dir_time);
   if (!status.IsOk()) {
     LOG_ERROR << "Failed to determine modification time for '" << model_dir_path
               << "': " << status.AsString();
-    return std::make_pair(0, 0);
+    return;
   }
+  model_timestamps_.emplace("", model_dir_time);
 
-  // Get files/folders in model_dir_path.
-  std::set<std::string> contents;
-  status = GetDirectoryContents(model_dir_path, &contents);
+  // Populate time for all immediate files/folders in 'model_dir_path'.
+  std::set<std::string> dir_contents;
+  status = GetDirectoryContents(model_dir_path, &dir_contents);
   if (!status.IsOk()) {
     LOG_ERROR << "Failed to determine modification time for '" << model_dir_path
               << "': " << status.AsString();
-    return std::make_pair(0, 0);
-  }
-  // Get latest modification time for each files/folders, and place it at the
-  // correct category.
-  for (const auto& child : contents) {
-    const auto full_path = JoinPath({model_dir_path, child});
-    if (full_path == model_config_path) {
-      // config.pbtxt or customized config file in configs folder
-      mtime.first = GetModifiedTime(full_path);
-    } else {
-      // model files
-      mtime.second = std::max(mtime.second, GetModifiedTime(full_path));
+    model_timestamps_.clear();
+    return;
```

Comment:
raise exception and try catch?

Reply:
Did a quick test on L0_lifecycle, the test currently expects any error related to retrieving timestamp to remain hidden and simply set the failed timestamp to 0. To translate it into this `ModelTimestamp` object, setting timestamp to 0 is the same as initializing the object using the default constructor, which is leaving the `model_timestamps_` object empty. Thus, if an exception is raised here, the caller catching the exception will simply re-initialize `ModelTimestamp` object using the default constructor (to remain aligned with the current logic of setting the timestamp to 0). In this case, raising an exception is redundant and can be handled by the constructor in one or two lines, or with a helper function specifically for keeping everything empty.

Another approach is we can change the test cases to accept the behavior change that failure to determine timestamp will fail the model load.

Reply:
I think to get this feature out quickly, we can retain the existing error handling on reading timestamps, and file another ticket for enhancing the error handling at a later time. @GuanLuo @rmccorm4 what do you think?

Reply:
Retaining existing error behavior and enhancing in a follow-up sounds fine to me.

Reply:
Sure, added the comment: [Comment on should raise an exception when failed to create timestamp](https://github.com/triton-inference-server/core/pull/388/commits/f73ab61dadc5b1f89ca5a5c3f41cd9fbb5db7db9)

Reply:
Can be addressed as follow-up, but I do think an exception should be raised, and it is fine to have the caller catch the exception and then initialize a timestamp with default constructor (if not failing). This is from the stand point where if the user initializes a timestamp with a path, the user is expecting the returned timestamp to carry correct timestamp w.r.t. the given path, not 0. And thus an error should be raised programmatically and it is up to the user to decide how to handle the error.